pkg/daemon/config.go

```diff
@@ -238,6 +255,31 @@ func getDefaultGatewayIface() (string, error) {
 	return "", errors.New("unable to find default route")
 }
 
+func getIfaceOwnPodIP(podIP string) (string, error) {
+	routes, err := netlink.RouteList(nil, syscall.AF_INET)
```

Comment:
you can use `netlink.LinkList` and `netlink.AddrList` to retrieve the ip address of each interface. The route way might be incorrect if no `src` field is set

Reply:
Good idea.